source/Util/DefaultHttpClientProvider.ts

```diff
@@ -0,0 +1,44 @@
+import { IHttpClientProvider } from './IHttpClientProvider';
+import { ApiCredentials, BearerTokenCredentials, IAMCredentials } from '@adastradev/serverless-discovery-sdk';
+import { IHttpClient } from './IHttpClient';
+
+const apigClientFactory: any = require('aws-api-gateway-client').default; // tslint:disable-line
+
+export class DefaultHttpClientProvider implements IHttpClientProvider {
```

Comment:
This will likely get repeated elsewhere as this thing evolves and moving it all into a class makes things reusable and testable.